documentation/config_file_auth_migration.md

```diff
@@ -0,0 +1,267 @@
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+            }
+            allow:
+            sort-order: 1
+            name:
+        },
+    ...
+    ```
+
+3. The legacy rule has a [`method`](./config_file_auth.html#method-1) setting, with an indirector value of `find` that's equivalent to the GET and POST HTTP methods. We can implement these by adding an optional HOCON [`method`](./config_file_auth.html#match-request) setting in the rule's `match-request` section and specifying GET and POST as an array.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow:
+            sort-order: 1
+            name:
+        },
+    ...
+    ```
+
+4. Next, set the [`allow`](#allow-allow-unauthenticated-and-deny) setting. The legacy rule used a `*` glob, which is also supported in HOCON.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow: *
+            sort-order: 1
+            name:
+        },
+    ...
+    ```
+
+5. Finally, give the rule a unique [`name`](#name) value. Remember that the rule will appear in logs and in the body of error responses to unauthorized clients.
+
+    ``` hocon
+    ...
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow: *
+            sort-order: 1
+            name: "environments"
+        },
+    ...
+    ```
+
+Our HOCON `auth.conf` file should now allow all clients to GET and POST to the `/puppet/v3/environments` endpoint, and should look like this:
+
+``` hocon
+authorization: {
+    version: 1
+    rules: [
+        {
+            match-request: {
+                path: /puppet/v3/environments
+                type: path
+                method: [get, post]
+            }
+            allow: *
+            sort-order: 1
+            name: "environments"
+        },
+    ]
+}
+```
+
+### Converting more complex rules
+
+#### Paths set by regular expressions
+
+To convert a regular expression path, enclose it in double quotation marks and solidus characters (`/`), and set the `type` to regex.
+
+> **Note:** You must escape regular expressions to conform to HOCON standards, which are the same as JSON's and differ from the deprecated format's regular expressions. For instance, the digit-matching regular expression `\d` must be escaped with a second backslash, as `\\d`.
+
+Deprecated:
+
+```
+path ~ ^/puppet/v3/catalog/([^/]+)$
+```
+
+HOCON:
+
+``` hocon
+...
+        match-request: {
+            path: "/^\/puppet\/v3\/catalog\/([^\/]+)$/"
```

Comment:
I don't think you need the leading and trailing solidus characters, and escaping the forward slash characters is optional in HOCON/JSON, I believe.  We just do this in the default catalog rule embedded in the Puppet Server package:

```
path: "^/puppet/v3/catalog/([^/]+)$"
```